api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -118,8 +120,22 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
     // TODO: check Google group membership to ensure user is in registered user group
 
     String userEmail = userInfo.getEmail();
+    WorkbenchConfig workbenchConfig = workbenchConfigProvider.get();
+    if (workbenchConfig.auth.serviceAccountApiUsers.contains(userEmail)) {
+      // Whitelisted service accounts are able to make API calls, too.
+      // TODO: stop treating service accounts as normal users, have a separate table for them,
+      // administrators.
+      User user = userDao.findUserByEmail(userEmail);
+      if (user == null) {
+        user = userService.createUser(null, null, userEmail, null);
```

Comment:
opt_nit: add new method `userService.createServiceAccountUser(userEmail)`? Looking at this callsite, passing a bunch of nulls seems like it could cause problems, would feel better if the logic were contained in UserService (even if we effectively wind up making the same call as here).

Reply:
Done.